editor/src/messages/portfolio/document/node_graph/document_node_types.rs

```diff
@@ -2253,6 +2260,9 @@ fn static_nodes() -> Vec<DocumentNodeDefinition> {
 				DocumentInputType::none(),
 				DocumentInputType::value("Size X", TaggedValue::F64(100.), false),
 				DocumentInputType::value("Size Y", TaggedValue::F64(100.), false),
+				DocumentInputType::value("Individual Corner Rounding", TaggedValue::Bool(false), false),
+				DocumentInputType::value("Corner Radius", TaggedValue::F64(0.), false),
```

Comment:
It is somewhat confusing to have the state encoded both in the type of the tagged value (either `TaggedValue::F64(_)` or `TaggedValue::F64Array4(_)`) and in this bool `"Individual Corner Rounding"`. Perhaps, in order to reduce chance of data inconsistency, you could remove this redundancy.

Reply:
A common way of doing this would be to use an enum:
```rust
enum CornerRounding {
   Individual([f64; 4),
   Uniform(f64),
)
```

Reply:
I don't think that is necessary as the typing is already stored in the TaggedValue enum @TrueDoctor. Introducing a new type adds complexity if we want the input to be exposed to the graph as we would have to add a convert node for the f64.

Reply:
+1 to @0HyperCube's reasoning.